tools/free_energy/alchemical_analysis/alchemical_analysis.xml

```diff
@@ -0,0 +1,95 @@
+<tool id="alchemical_analysis" name="Alchemical Analysis" version="1.0.2">
+    <description>Analysis of alchemical free energy calculations</description> 
+    <requirements>
+         <requirement type="package" version="1.0.2">alchemical-analysis</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+       ln -s '$datain' ./data.tar &&
+       tar xf ./data.tar && 
+       mkdir ./RESULTS/ &&
+            #if $lambda.skip_lambda == "no"
+                alchemical_analysis -d data -o ./RESULTS -t '$temp' -s '$time' -c -q xvg -p '$prefix' -g -e -f 10 -w -u '$units' &>> '$output'
+            #else if $lambda.skip_lambda == "yes"
+                 alchemical_analysis -d data -o ./RESULTS -t '$temp' -s '$time' -k '$lambda.skiplambda' -c -q xvg -p '$prefix' -g -e -f 10 -w -u '$units' &>> '$output'
+            #end if
+    ]]></command>
+    <inputs>
+    <param format="tar" name="datain" type="data" label="TI/FEP data output"/>
+    <param name="prefix" type="text" value="md" label="Prefix for datafile sets" help="This should be md if you used the alchemical simulator tool"/>
+    <param name="temp" type="integer" value="300" label="Temperature (K)"/>
+    <param name="time" type="integer" value="0" label="Equilibration time (ps)" help="Discard data prior to this specified time as 'equilibration' data." />
+    <param name="units" type="select" label="Units to report energies" >
+        <option value="kcal">kcal</option>
+        <option value="kJ">kJ</option>
+        <option value="kBT">kT</option>
+    </param>
+    <conditional name="lambda">
+        <param name="skip_lambda" type="select" label="Skip some lambda states?"  >
+            <option value="no">No</option>
+            <option value="yes">Yes</option>
+        </param>
+            <when value="no">
+            </when>
+            <when value="yes">
+                <param name="skiplambda" type="text" value="1-2-3" label="Indices of lambda states to skip" help="Give lambda indices separated by '-' and they will be removed from the analysis."/>
```

Comment:
Why is the conditional here needed? If the value is empty and optional="true" then the conditional is not needed, isn't it?

Reply:
@bgruening I wanted not to show this option by default If wrong values are specified the tool will not run. Should I show this option by default and make it blank?

Reply:
What are correct values and what are wrong, is there a patter that you can check? Galaxy supports "validators", to validate a user-input.

Reply:
@tsenapathi, see in my PR for an example: https://github.com/galaxycomputationalchemistry/galaxy-tools-compchem/pull/44/files#diff-a4db2c2bf23fc0cec5379f8b9d9a92d6R21

Reply:
I see. However, if I want to use the argument for skipping lambda (which is -k) I need to specify some parameters, white space is not allowed in my case. That's why I had to use two separate options.

Reply:
You can can change the validator to accept the input you want. See: https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-inputs-param-validator